indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,534 @@
+            var rooms = {{ rooms | tojson }};
+            var hiddenData = field.val() ? JSON.parse(field.val()) : {};
+            var latestUsedField = '';
+            var latestSelectedOption = {field: '', value: ''};
+            var latestGroup = '';
+            var currentScrollPos = 0;
+            var delta = 0;
+            var validRoom = false;
+            var validVenue = false;
+            var preventGuessing = false;
+            var roomJustCleared = false;
+            var venueJustChanged = false;
+            var justUseOptionSelected = false;
+            var resultsWereOpen = false;
+
+            function initializeInputs() {
+                address.val(hiddenData.address);
+                roomInput.val(hiddenData.room_name);
+                venueInput.val(hiddenData.venue_name);
+                if (hiddenData.venue_id) {
+                    highlightOption(venueInput, venueInput.val());
+                }
+                if (hiddenData.room_id) {
+                    highlightOption(roomInput, roomInput.val());
+                }
+                usingDefault.prop('checked', hiddenData.inheriting);
+            }
+
+            function toggleDefaultLocation(isChecked, isInitStep) {
+                var message = ($T.gettext('Using default location from: ') + {{ parent[1] | tojson }} +
+                               ' (' + {{ parent[0] | tojson }} + ')');
+                var text = isChecked ? message : $T.gettext('Using custom location');
+                description.html(text);
+                venueInput.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+                roomInput.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+                address.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+                field.closest('.i-location-field').toggleClass('disabled', isChecked);
+
+                hiddenData.inheriting = isChecked;
+                field.val(JSON.stringify(hiddenData)).trigger('change');
+
+                if (!isInitStep) {
+                    if (isChecked) {
+                        hiddenData = {{ field.data.source.location_data_for_ui | tojson }}
+                        hiddenData.inheriting = true;
+                        field.val(JSON.stringify(hiddenData)).trigger('change');
+                        initializeInputs();
+                    } else {
+                        clearVenue();
+                    }
+                }
+            };
+
+            function reorderCategories(venueName) {
+                var groupOrder = [venueName];
+                for (var i = venueNames.length - 1; i >= 0; i--) {
+                    if (!~groupOrder.indexOf(venueNames[i])) {
+                        groupOrder.push(venueNames[i]);
+                    }
+                }
+                Typeahead[roomInput.selector].options.groupOrder = groupOrder;
```

Comment:
I thought there was a better way to access a specific element's typeahead instance??